be/src/exec/pipeline/scan/scan_operator.h

```diff
@@ -36,6 +36,8 @@ class ScanOperator : public SourceOperator {
     void set_workgroup(workgroup::WorkGroupPtr wg);
 
     // interface for different scan node
+    virtual bool maybe_has_output() const { return true; }
+    virtual bool must_be_finished() const { return false; }
```

Comment:
Leave some comment? What does `must_be_finished` means?

Reply:
It is to change the behaviour of `has_output` and `is_finished` by the child `OlapScanOpeartor`. 
However, it is hard to understand, so I remove them, and make `OlapScanOperator` overrides `has_output` and `is_finished`.